packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -171,15 +172,38 @@ abstract contract ForwarderModule is BaseModule
                 metaTx.to == controller().walletFactory()
             );
 
+            // MAX_REIMBURSTMENT_OVERHEAD covers an ERC20 transfer and a quota update.
+            if (skipQuota) {
+                gasUsed -= 48000;
+            }
+
+            if (metaTx.gasToken == address(0)) {
+                gasUsed -= 15000; // diff between an regular ERC20 transfer and an ETH send
```

Comment:
These gas calculations depend on a lot of assumptions. Wouldn't it be easier just to let the user sign a fixed gas overhead cost which we could accurately calculate offchain so we don't have to do try and calculate this onchain?

Reply:
I like the idea.